hca/define_api.py

```diff
@@ -1,37 +1,209 @@
 from __future__ import absolute_import, division, print_function, unicode_literals
 
-import json, sys, os
-from io import open
+import json
+import os
+import sys
 import requests
-from .parser import get_parser
+
+from io import open
+
+from .parser import get_parser, ADDED_COMMANDS
+from .constants import Constants
 
 
 class API:
     """Class for interacting with the REST Api."""
 
-    def __init__(self, api_url, user):
+    def __init__(self, use_test_api=False):
         """Initialize the CLI API."""
-        self.spec = self.get_spec()
-        self.base_url = self.spec['host'] + self.spec['basePath']
-        self.parser = get_parser(self.spec)
+        self.use_test_api = use_test_api
+        spec = self.get_spec()
+        scheme = spec['schemes'][0] if 'schemes' in spec else "https"
+        self.base_url = "{}://{}{}".format(scheme, spec['host'], spec['basePath'])
+        self.parser, self.param_data = get_parser(spec)
 
     def get_spec(self):
-        """Load the API specification."""
-        with open(os.path.dirname(os.path.realpath(__file__)) + "/api_spec.json") as fp:
+        """
+        Load the API specification.
+
+        self.use_test_api is a boolean flag to indicate which spec to use.
+            api_spec is the spec downloaded from the api on build.
+            ../test/test is the mocked up spec I've been toying with.
```

Comment:
Definitely a bit wonky (and looking through git history the way I had originally done it). I could call `os.path.dirname` again before, but I feel like that would be even messier - pretty much everyone understands what two dots means.